File: launchdarkly/csharp/src/OSEG.LaunchDarklyExamples/UpdatePhaseStatusExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Org.LaunchDarklyTools.Api;
using Org.LaunchDarklyTools.Client;
using Org.LaunchDarklyTools.Model;

namespace OSEG.LaunchDarklyExamples;

public class UpdatePhaseStatusExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey.Add("Authorization", "YOUR_API_KEY");

        var audiences1ReleaseGuardianConfiguration = new ReleaseGuardianConfigurationInput(
            monitoringWindowMilliseconds: 60000,
            rolloutWeight: 50,
            rollbackOnRegression: true,
            randomizationUnit: "user"
        );

        var audiences1 = new ReleaserAudienceConfigInput(
            notifyMemberIds: [
                "1234a56b7c89d012345e678f",
            ],
            notifyTeamKeys: [
                "example-reviewer-team",
            ],
            releaseGuardianConfiguration: audiences1ReleaseGuardianConfiguration
        );

        var audiences = new List<ReleaserAudienceConfigInput>
        {
            audiences1,
        };

        var updatePhaseStatusInput = new UpdatePhaseStatusInput(
            audiences: audiences
        );

        try
        {
            var response = new ReleasesBetaApi(config).UpdatePhaseStatus(
                projectKey: "projectKey_string",
                flagKey: "flagKey_string",
                phaseId: "phaseId_string",
                updatePhaseStatusInput: updatePhaseStatusInput
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling ReleasesBetaApi#UpdatePhaseStatus: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: launchdarkly/typescript-axios/src/UpdatePhaseStatusExample.ts
import axios, { AxiosError } from "axios";
import * as api from "launchdarkly_client"

const configuration = new api.Configuration({
  apiKey: "YOUR_API_KEY",
});

const audiences1ReleaseGuardianConfiguration: api.ReleaseGuardianConfigurationInput = {
  monitoringWindowMilliseconds: 60000,
  rolloutWeight: 50,
  rollbackOnRegression: true,
  randomizationUnit: "user",
};

const audiences1: api.ReleaserAudienceConfigInput = {
  notifyMemberIds: [
    "1234a56b7c89d012345e678f",
  ],
  notifyTeamKeys: [
    "example-reviewer-team",
  ],
  releaseGuardianConfiguration: audiences1ReleaseGuardianConfiguration,
};

const audiences = [
  audiences1,
];

const updatePhaseStatusInput: api.UpdatePhaseStatusInput = {
  audiences: audiences,
};

new api.ReleasesBetaApi(configuration).updatePhaseStatus(
  "projectKey_string", // projectKey
  "flagKey_string", // flagKey
  "phaseId_string", // phaseId
  updatePhaseStatusInput,
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling ReleasesBetaApi#updatePhaseStatus:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
